### PR TITLE
Fix issues from pull #20

### PR DESCRIPTION
**Pull request by margaretchan**
This is the exact same as pull request #20 with 1 more commit addressing the comments. I accidentally merged too early and can't figure out how to re-open the pull request. This also reverts the old revert.

### CODE REVIEW

**Review by josefj1519 — COMMENTED** [review]
Delete PromoCodeScraper.java and  PromoCodeScraperTest.java from this PR.  I reverted those files.

File: src/main/java/com/google/step/youtube/Channel.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import com.google.auto.value.AutoValue;
+import java.util.Date;
+
+/**
+ * Value type for a YouTube channel and information on its videos. 
+ */
+@AutoValue
+abstract class Channel {
+
+    public static Channel create (String channelUrl, String uploadPlaylistId, Date lastVideoDate,
```
Comment (josefj1519) [inline src/main/java/com/google/step/youtube/Channel.java]: channelId

Comment (margaretchan) [reply src/main/java/com/google/step/youtube/Channel.java]: Changed.

**Review by josefj1519 — APPROVED** [review]
Delete PromoCodeScraper.java and  PromoCodeScraperTest.java from this PR.  I reverted those files.

**Review by dallen68 — CHANGES_REQUESTED** [review]
Add PromoCodeSevelet test

**Review by dallen68 — APPROVED** [review]

File: src/main/java/com/google/step/youtube/DescriptionParser.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+/**
+ * Class for parsing the descriptions of YouTube videos for promotional codes and affiliate links. 
+ */
+public class DescriptionParser {
+
```
Comment (derrillhd) [inline src/main/java/com/google/step/youtube/DescriptionParser.java]: super nit: say implementation is coming.

File: src/test/java/com/google/step/servlets/PromoCodeServletTest.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.google.step.servlets;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+// @RunWith(JUnit4.class)
+public final class PromoCodeServletTest {
+
```
Comment (derrillhd) [inline src/test/java/com/google/step/servlets/PromoCodeServletTest.java]: nit: add comment todo saying implementation is coming.

**Review by derrillhd — APPROVED** [review]

File: src/main/java/com/google/step/youtube/DescriptionParser.java (hunk 2 above)
Comment (margaretchan) [reply src/main/java/com/google/step/youtube/DescriptionParser.java]: Added.

File: src/test/java/com/google/step/servlets/PromoCodeServletTest.java (hunk 3 above)
Comment (margaretchan) [reply src/test/java/com/google/step/servlets/PromoCodeServletTest.java]: Added.

**Pull request merged**